test/functional/tests/cli/test_zero_metadata_command.py

```diff
@@ -0,0 +1,258 @@
+    with TestRun.step("Run zero metadata."):
+        output = TestRun.executor.run(cli.zero_metadata_cmd(str(cache_dev.path)))
+        if output.exit_code == 0:
+            TestRun.LOGGER.info("Zeroing metadata ran successfully!")
+        else:
+            TestRun.LOGGER.error("Zero metadata should works for cache device after cache stop!")
+
+    with TestRun.step("Load cache."):
+        try:
+            cache = casadm.load_cache(cache_dev)
+        except CmdException:
+            TestRun.LOGGER.info("Load cache failed as expected.")
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+@pytest.mark.parametrizex("filesystem", Filesystem)
+def test_zero_metadata_filesystem(filesystem):
+    """
+        title: Test Test for '--zero-metadata' and filesystem.
+        description: |
+          Test for '--zero-metadata' on drive with filesystem.
+        pass_criteria:
+          - Zero metadata on device with filesystem failed and not removed filesystem.
+          - Zero metadata on mounted device failed.
+    """
+    mount_point = "/mnt"
+    with TestRun.step("Prepare devices."):
+        cache_disk, core_disk = prepare_devices()
+
+    with TestRun.step("Start cache and add core."):
+        cache = casadm.start_cache(cache_disk, force=True)
+        core = cache.add_core(core_disk)
+
+    with TestRun.step("Create filesystem on core device."):
+        core.create_filesystem(filesystem)
+
+    with TestRun.step("Zero metadata on core device and check if filesystem still exists"):
+        output = TestRun.executor.run(cli.zero_metadata_cmd(str(core.path)))
+        if output.exit_code != 0:
+            cli_messages.check_stderr_msg(output, cli_messages.no_cas_metadata)
+        else:
+            TestRun.LOGGER.error("Zeroing metadata should fail!")
+
+        file_system = get_device_filesystem_type(core.get_device_id())
+
+        if file_system != filesystem:
+            TestRun.LOGGER.error(f"After zero metadata on core - filesystem ({file_system}) "
+                                 f"not as expected ({filesystem})!")
+
+    with TestRun.step("Mount core."):
+        core.mount(mount_point)
+
+    with TestRun.step("Zero metadata on mounted core device and validate result"):
+        output = TestRun.executor.run(cli.zero_metadata_cmd(str(core.path)))
+        if output.exit_code != 0:
+            cli_messages.check_stderr_msg(output, cli_messages.unavailable_device)
+        else:
+            TestRun.LOGGER.error("Zeroing metadata should fail!")
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_zero_metadata_dirty_data():
+    """
+        title: Test for '--zero-metadata' and dirty data scenario.
+        description: |
+          Test for '--zero-metadata' with&without 'force' option if there are dirty data on cache.
```

Comment:
```suggestion
          Test for '--zero-metadata' with & without 'force' option if there are dirty data on cache.
```
Or even `with and without`